airbyte-integrations/connectors/source-mssql/src/test/java/io/airbyte/integrations/source/mssql/CdcStateCompressionTest.java

```diff
@@ -0,0 +1,254 @@
+
+  static private final String CDC_ROLE_NAME = "cdc_selector";
+
+  static private final String TEST_USER_NAME_PREFIX = "cdc_test_user";
+
+  static private final String TEST_SCHEMA = "test_schema";
+
+  static private final int TEST_TABLES = 10;
+
+  static private final int ADDED_COLUMNS = 1000;
+
+  static private final MSSQLServerContainer<?> CONTAINER = new MsSQLContainerFactory().shared(
+      "mcr.microsoft.com/mssql/server:2022-latest", "withAgent");
+
+  private MsSQLTestDatabase testdb;
+
+  @BeforeEach
+  public void setup() {
+    testdb = new MsSQLTestDatabase(CONTAINER);
+    testdb = testdb
+        .withConnectionProperty("encrypt", "false")
+        .withConnectionProperty("databaseName", testdb.getDatabaseName())
+        .initialized()
+        .withSnapshotIsolation()
+        .withCdc()
+        .withWaitUntilAgentRunning();
+
+    // Create a test schema and a bunch of test tables with CDC enabled.
+    // Insert one row in each table so that they're not empty.
+    final var enableCdcSqlFmt = """
+                                EXEC sys.sp_cdc_enable_table
+                                \t@source_schema = N'%s',
+                                \t@source_name   = N'test_table_%d',
+                                \t@role_name     = N'%s',
+                                \t@supports_net_changes = 0,
+                                \t@capture_instance = N'capture_instance_%d_%d'
+                                """;
+    testdb.with("CREATE SCHEMA %s;", TEST_SCHEMA);
+    for (int i = 0; i < TEST_TABLES; i++) {
+      testdb
+          .with("CREATE TABLE %s.test_table_%d (id INT IDENTITY(1,1) PRIMARY KEY);", TEST_SCHEMA, i)
+          .with(enableCdcSqlFmt, TEST_SCHEMA, i, CDC_ROLE_NAME, i, 1)
+          .with("INSERT INTO %s.test_table_%d DEFAULT VALUES", TEST_SCHEMA, i);
+    }
+
+    // Create a test user to be used by the source, with proper permissions.
+    testdb
+        .with("CREATE LOGIN %s WITH PASSWORD = '%s', DEFAULT_DATABASE = %s", testUserName(), testdb.getPassword(), testdb.getDatabaseName())
+        .with("CREATE USER %s FOR LOGIN %s WITH DEFAULT_SCHEMA = [dbo]", testUserName(), testUserName())
+        .with("REVOKE ALL FROM %s CASCADE;", testUserName())
+        .with("EXEC sp_msforeachtable \"REVOKE ALL ON '?' TO %s;\"", testUserName())
+        .with("GRANT SELECT ON SCHEMA :: [%s] TO %s", TEST_SCHEMA, testUserName())
+        .with("GRANT SELECT ON SCHEMA :: [cdc] TO %s", testUserName())
+        .with("USE [master]")
+        .with("GRANT VIEW SERVER STATE TO %s", testUserName())
+        .with("USE [%s]", testdb.getDatabaseName())
+        .with("EXEC sp_addrolemember N'%s', N'%s';", CDC_ROLE_NAME, testUserName());
+
+    // Increase schema history size to trigger state compression.
+    // We do this by adding lots of columns with long names,
+    // then migrating to a new CDC capture instance for each table.
+    // This is admittedly somewhat awkward and perhaps could be improved.
```

Comment:
I'm holding out until I better understand how Debezium + SQL Server do CDC. Perhaps I'm overlooking something obvious. Otherwise yes absolutely let's trigger the compression sooner.